### PR TITLE
feature/547_occur_aggregation_sth

### PR DESCRIPTION
**Pull request by frbattid**
* Implements issue #547 
* 100% unit tests passed:
```
Tests run: 80, Failures: 0, Errors: 0, Skipped: 0
```
* (unofficial) e2e tests passed:
```
time=2016-03-14T11:48:37.415CET | lvl=INFO | trans=1457952512-624-0000000000 | svc=test | subsvc=sthoccur | function=getEvents | comp=Cygnus | msg=com.telefonica.iot.cygnus.handlers.OrionRestHandler[263] : Received data ({  "subscriptionId" : "51c0ac9ed714fb3b37d7d5a8",  "originator" : "localhost",  "contextResponses" : [    {      "contextElement" : {        "attributes" : [          {            "name" : "schedule",            "type" : "json",            "value" : "this is a string"          }        ],        "type" : "Events",        "isPattern" : "false",        "id" : "thisweek"      },      "statusCode" : {        "code" : "200",        "reasonPhrase" : "OK"      }    }  ]})
time=2016-03-14T11:49:41.099CET | lvl=INFO | trans=1457952512-624-0000000001 | svc=test | subsvc=sthoccur | function=getEvents | comp=Cygnus | msg=com.telefonica.iot.cygnus.handlers.OrionRestHandler[263] : Received data ({  "subscriptionId" : "51c0ac9ed714fb3b37d7d5a8",  "originator" : "localhost",  "contextResponses" : [    {      "contextElement" : {        "attributes" : [          {            "name" : "schedule",            "type" : "json",            "value" : "this is a string"          }        ],        "type" : "Events",        "isPattern" : "false",        "id" : "thisweek"      },      "statusCode" : {        "code" : "200",        "reasonPhrase" : "OK"      }    }  ]})
time=2016-03-14T11:50:00.691CET | lvl=INFO | trans=1457952512-624-0000000002 | svc=test | subsvc=sthoccur | function=getEvents | comp=Cygnus | msg=com.telefonica.iot.cygnus.handlers.OrionRestHandler[263] : Received data ({  "subscriptionId" : "51c0ac9ed714fb3b37d7d5a8",  "originator" : "localhost",  "contextResponses" : [    {      "contextElement" : {        "attributes" : [          {            "name" : "schedule",            "type" : "json",            "value" : "this is a"          }        ],        "type" : "Events",        "isPattern" : "false",        "id" : "thisweek"      },      "statusCode" : {        "code" : "200",        "reasonPhrase" : "OK"      }    }  ]})
..
..
{ "_id" : { "attrName" : "schedule", "origin" : ISODate("2016-01-01T00:00:00Z"), "resolution" : "month", "range" : "year" }, "points" : [ { "offset" : 1, "samples" : 0, "occur" : {  } }, { "offset" : 2, "samples" : 0, "occur" : { "this is a string" : 2, "this is a" : 1 } }, { "offset" : 3, "samples" : 3, "occur" : {  } }, { "offset" : 4, "samples" : 0, "occur" : {  } }, { "offset" : 5, "samples" : 0, "occur" : {  } }, { "offset" : 6, "samples" : 0, "occur" : {  } }, { "offset" : 7, "samples" : 0, "occur" : {  } }, { "offset" : 8, "samples" : 0, "occur" : {  } }, { "offset" : 9, "samples" : 0, "occur" : {  } }, { "offset" : 10, "samples" : 0, "occur" : {  } }, { "offset" : 11, "samples" : 0, "occur" : {  } }, { "offset" : 12, "samples" : 0, "occur" : {  } } ], "attrType" : "json" }
```
* Assignee @pcoello25

### CODE REVIEW

File: doc/flume_extensions_catalogue/orion_sth_sink.md [hunk 1]
```diff
@@ -17,7 +17,18 @@ Content:
...
 ##<a name="section1"></a>Functionality
-`com.iot.telefonica.cygnus.sinks.OrionSTHSink`, or simply `OrionSTHSink` is a sink designed to persist NGSI-like context data events within a MongoDB server in an aggregated way. Usually, such a context data is notified by a [Orion Context Broker](https://github.com/telefonicaid/fiware-orion) instance, but could be any other system speaking the <i>NGSI language</i>.
+`com.iot.telefonica.cygnus.sinks.OrionSTHSink`, or simply `OrionSTHSink` is a sink designed to persist NGSI-like context data events within a MongoDB server in an aggregated way, specifically these measures are computed:
+
+* For numeric attribute values:
+    * Sum of all the samples.
+    * Sum of the square value of all the samples.
+    * Maximum value among all the samples.
+    * Minimum value among all the samples.
+* Number of occurences for string attribute values.
```
Comment (pcoello25) [inline doc/flume_extensions_catalogue/orion_sth_sink.md]: occurences -> occurrences

File: README.md [hunk 2]
```diff
@@ -269,11 +269,12 @@ Detailed information regarding Cygnus can be found in the [Installation and Admi
   <tr><td>Time and size-based data management policies</td><td>0.13.0</td></tr>
   <tr><td rowspan="2">OrionMySQLSink</td><td>First implementation</td><td>0.2.0</td></tr>
   <tr><td>Batching mechanism</td><td>0.10.0</td></tr>
-  <tr><td rowspan="5">OrionSTHSink</td><td>First implementation</td><td>0.8.0</td></tr>
+  <tr><td rowspan="6">OrionSTHSink</td><td>First implementation</td><td>0.8.0</td></tr>
   <tr><td>Hash based collections</td><td>0.8.1</td></tr>
   <tr><td>TimeInstant metadata as reception time</td><td>0.12.0</td></tr>
   <tr><td>Batching mechanism</td><td>0.13.0</td></tr>
   <tr><td>Time and size-based data management policies</td><td>0.13.0</td></tr>
+  <tr><td>String-based aggregation (ocurrences)</td><td>0.14.0</td></tr>
```
Comment (pcoello25) [inline README.md]: ocurrences -> occurrences

**Comment by pcoello25** [comment]
LGTM

File: doc/flume_extensions_catalogue/orion_sth_sink.md (hunk 1 above)
Comment (frbattid) [inline doc/flume_extensions_catalogue/orion_sth_sink.md]: Fixed in https://github.com/telefonicaid/fiware-cygnus/commit/2d262bc5171759041c2b93dda3cc088c299d32d2

File: README.md (hunk 2 above)
Comment (frbattid) [inline README.md]: Fixed in https://github.com/telefonicaid/fiware-cygnus/commit/2d262bc5171759041c2b93dda3cc088c299d32d2

**Pull request merged**